packages/preview/lib/preview.js

```diff
@@ -104,6 +105,42 @@ const expire = async (requests, users, pgdb, mail) => {
   )
 }
 
+const followup = async (requests, users, memberships, pgdb, t) => {
+  debug('followup', {
+    requests: requests.length,
+    users: users.length,
+    memberships: memberships.length
+  })
+
+  await Promise.map(
+    requests,
+    async (request) => {
+      const transaction = await pgdb.transactionBegin()
+
+      try {
+        const user = users.find(user => user.id === request.userId)
+        const hasMembership =
+          !!memberships.find(membership => membership.userId === user.id)
+
+        await setFollowup(request, user, pgdb)
+
+        if (!hasMembership) {
+          await mailLib.sendFollowup({ user, request, pgdb: transaction, t })
+        }
+
+        await transaction.transactionCommit()
+      } catch (e) {
+        await transaction.transactionRollback()
+
+        debug('rollback', { request: request.id })
+
+        throw e
+      }
+    },
+    { concurrency: 5 }
+  )
```

Comment:
Can one user have multiple open requests?

Reply:
A user can have multiple subsequent requests. A user can resubmit a request when other request is expired.

Reply:
Ok, so this is not going to send multiple emails to one user, because each user can only have one request where `expireAt` is null?

Reply:
Yes.

Starts [with `begin` function](https://github.com/orbiting/backends/blob/da8bec239f0093ddbe243dba59f99c8773dfca72/packages/preview/lib/preview.js#L38-L42) which queries [for "other" (valid) request](https://github.com/orbiting/backends/blob/da8bec239f0093ddbe243dba59f99c8773dfca72/packages/preview/lib/preview.js#L175-L180) before adding request.